discord/commands/mixins.py

```diff
@@ -0,0 +1,910 @@
+    def author(self) -> User | Member:
+        """Union[:class:`.User`, :class:`.Member`]: Returns the author associated with this context's command."""
+        return self.source.author
+
+    @property
+    def user(self) -> User | Member:
+        """Union[:class:`.User`, :class:`.Member`]: Alias for :attr:`BaseContext.author`."""
+        return self.author
+
+    @utils.cached_property
+    def me(self) -> Member | ClientUser:
+        """Union[:class:`.Member`, :class:`.ClientUser`]:
+        Similar to :attr:`.Guild.me` except it may return the :class:`.ClientUser` in private message
+        message contexts, or when :meth:`.Intents.guilds` is absent.
+        """
+        # bot.user will never be None at this point.
+        return self.guild.me if self.guild and self.guild.me else self.bot.user  # type: ignore
+
+    @property
+    def voice_client(self) -> VoiceProtocol | None:
+        r"""Optional[:class:`.VoiceProtocol`]: A shortcut to :attr:`.Guild.voice_client`\, if applicable."""
+        return self.guild.voice_client if self.guild else None
+
+
+class Invokable(Generic[CogT, P, T]):
+    r"""A baseclass to provide ***basic & common functionality*** between
+    :class:`.ApplicationCommand` and :class:`~ext.commands.Command`.
+
+    .. versionadded:: 2.2
+
+    Attributes
+    ----------
+    name: str
+        The name of the invokable/command.
+    callback: :ref:`coroutine <coroutine>`
+        The coroutine that is executed when the command is called.
+    parent: Optional[:class:`Invokable`]
+        The parent group of this command.
+    cog: Optional[:class:`Cog`]
+        The cog that this command belongs to. ``None`` if there isn't one.
+    enabled: :class:`bool`
+        A boolean that indicates if the command is currently enabled.
+        If the command is invoked while it is disabled, then
+        :exc:`.DisabledCommand` is raised to the :func:`.on_command_error`
+        event. Defaults to ``True``.
+    checks: List[Callable[[:class:`.BaseContext`], :class:`bool`]]
+        A list of predicates that verifies if the command could be executed with the given
+        :class:`.BaseContext` (:class:`.ApplicationContext` or :class:`~ext.commands.Context`
+        to be specific) as the sole parameter. If an exception is necessary to be thrown to
+        signal failure, then one inherited from :exc:`.CommandError` should be used. Note that
+        if the checks fail then :exc:`.CheckFailure` exception is raised to the :func:`.on_command_error`
+        event.
+    cooldown_after_parsing: :class:`bool`
+        If ``True``\, cooldown processing is done after argument parsing,
+        which calls converters. If ``False`` then cooldown processing is done
+        first and then the converters are called second. Defaults to ``False``.
+    cooldown: Optional[:class:`Cooldown`]
+        The cooldown applied when the command is invoked.
+    """
+
+    def __init__(self, func: CallbackT, **kwargs):
```

Comment:
You should type the parameters expected in `kwargs` via an overload or just explicitly typing them as keyword only parameters.